### DISCUSSION TITLE
CaseFlowCylinder_Re200 DualSPHysica_5.2

### DISCUSSION THREAD

**Discussion by FisMatSPH** [DualSPHysics/DualSPHysics#202] [DUALSPHYSICS_V5.2]
Dear DualSPHysic users, greetings to all of you, here is part of my Master's work.

I made a 2D model and I have encountered an error similar to Shadi's, and I would like you to please help me. this is my model.
![mi modelo-inlet_outlet](https://github.com/DualSPHysics/DualSPHysics/assets/161859633/49221646-be38-4bd4-8925-728f4ad4f284)

WARNING: The option DDT:2 is equal to DDT:1 when gravity.z is zero.

WARNING: There are 197 of 377 fixed or moving boundary particles without normal data.

Error reading xml - Value of 'mkfluid' invalid. No particles data with mkfluid=1 to define inout points.

File: CaseFlowCylinder_Re200_out/CaseFlowCylinder_Re200.xml(row:116)

Finished execution (code=1).

and here is how to declare my geometry and my fluid area:

<hdp value="2.1" comment="Alternative option to calculate the smoothing length (h=hdp*dp)" />

      <cflnumber value="0.2" comment="Coefficient to multiply dt" />

    </constantsdef>

    <mkconfig boundcount="240" fluidcount="4" />

    <geometry>

      <definition dp="0.00020">

        <pointmin x="0" y="0" z="0" />   

        <pointmax x="0.012" y="0" z="0.0041" /> 

      </definition>

      <commands>

        <mainlist>

          <setshapemode>dp | bound</setshapemode>

          <setdrawmode mode="full" />

          

          <setmkfluid mk="0" />

          <drawbox>

            <boxfill>solid</boxfill>

            <point x="0" y="0.00001" z="0.0002" />

            <size x="0.012" y="0.00001" z="0.0037" /> 

          </drawbox>

          

          <setmkfluid mk="1" />

          <drawbox>

            <boxfill>left</boxfill>

            <point x="0" y="0.00001" z="0.0002" />

            <size x="0.012" y="0.00001" z="0.0037" />

          </drawbox>

          <setmkfluid mk="2" />

          <drawbox>

            <boxfill>right</boxfill>

            <point x="0" y="0.00001" z="0.0002" />

            <size x="0.012" y="0.00001" z="0.0037" /> 

          </drawbox>

          

          <setmkbound mk="0" />

          <drawcylinder radius="0.001100">

            <point x="0.004" y="-0.001300" z="0.003175" />

            <point x="0.008" y="0.001300" z="0.003185" />

          </drawcylinder>

          

          <setmkbound mk="1" />

          <drawbox>

            <boxfill>solid</boxfill>

            <point x="0" y="0" z="0" />

            <size x="0.012" y="0.0001" z="0.0002" />

          </drawbox>

          <setmkbound mk="2" />

          <drawbox>

            <boxfill>solid</boxfill>

            <point x="0" y="0" z="0.0039" />

            <size x="0.012" y="0.0010" z="0.0002" />

          </drawbox>

this is how the model looks like but the particles enter all out of order:

![mi modelo-inlet_outlet_en_paraview](https://github.com/DualSPHysics/DualSPHysics/assets/161859633/e6f8baa6-c503-4e5a-b80e-8541e7ab87bc)

I would be glad if you could help me to solve it.

Thank you.

**Comment by FisMatSPH** [comment]
I have already changed the fluid geomteria making it gradually smaller, then change the dp such that h/dp<1.5, then gradually change the initial velocity to 0.27 m/s. I have already changed the fluid geomteria making it gradually smaller, then change the dp such that h/dp<1.5, then gradually change the initial velocity to 0.27 m/s. I also gradually changed the number of layers in the buffer zone. All this still gives me the same error: 
WARNING: The DDT:2 option is equal to DDT:1 when gravity.z is zero.

WARNING: There are 197 of 377 stationary or moving boundary particles with no normal data.

Error reading xml - Invalid 'mkfluid' value. There is no particle data with mkfluid=1 to define entry points.
File: CaseFlowCylinder_Re200_out/CaseFlowCylinder_Re200.xml(row:116)
Execution completed (code=1).


I can't figure out what I have done wrong, so it doesn't run normally. 
I would appreciate your comments and suggestions.

**Comment by EnglishA1** [comment]
Hi 

You need to increase the size of yoour domian in the x direction in both the positive and negative regions. If you look at Case_FlowCylinder in the inletoutlet cases you will see what I mean. In that case the point min is at -1.5 and the solid domain start at -0.6 and the inlet is defined at -5.10, this gives room for the inlet buffer zone to be created inside the overall fluid domain. The same needs to be done at the outlet. To start I suggest you set your pointmin to x=-0.01 and point mas to x=0.022 and start your side walls from x=-0.005 with length 0.013 and see if that works. This should give you room for 25 layers of buffer particles (more than you need)

I hope this helps
